flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/Expressions.java

```diff
@@ -0,0 +1,562 @@
+	 * @param ifTrue expression to be evaluated if condition holds
+	 * @param ifFalse expression to be evaluated if condition does not hold
+	 */
+	public static ApiExpression ifThenElse(Object condition, Object ifTrue, Object ifFalse) {
+		return apiCall(BuiltInFunctionDefinitions.IF, condition, ifTrue, ifFalse);
+	}
+
+	/**
+	 * Creates an expression that selects a range of columns. It can be used wherever an array of
+	 * expression is accepted such as function calls, projections, or groupings.
+	 *
+	 * <p>A range can either be index-based or name-based. Indices start at 1 and boundaries are
+	 * inclusive.
+	 *
+	 * <p>e.g. in Scala: withColumns(range("b", "c")) or withoutColumns($("*"))
+	 */
+	public static ApiExpression withColumns(Object head, Object... tail) {
+		return apiCallAtLeastOneArgument(BuiltInFunctionDefinitions.WITH_COLUMNS, head, tail);
+	}
+
+	/**
+	 * Creates an expression that selects all columns except for the given range of columns. It can
+	 * be used wherever an array of expression is accepted such as function calls, projections, or
+	 * groupings.
+	 *
+	 * <p>A range can either be index-based or name-based. Indices start at 1 and boundaries are
+	 * inclusive.
+	 *
+	 * <p>e.g. withoutColumns(range("b", "c")) or withoutColumns($("c"))
+	 */
+	public static ApiExpression withoutColumns(Object head, Object... tail) {
+		return apiCallAtLeastOneArgument(BuiltInFunctionDefinitions.WITHOUT_COLUMNS, head, tail);
+	}
+
+	/**
+	 * A call to a function that will be looked up in a catalog. There are two kinds of functions:
+	 * <ul>
+	 *     <li>System functions - which are identified with one part names</li>
+	 *     <li>Catalog functions - which are identified always with three parts names
+	 *     (catalog, database, function)</li>
+	 * </ul>
+	 *
+	 * <p>Moreover each function can either be a temporary function or permanent one
+	 * (which is stored in an external catalog).
+	 *
+	 * <p>Based on that two properties the resolution order for looking up a function based on
+	 * the provided {@code functionName} is following:
+	 * <ul>
+	 *     <li>Temporary system function</li>
+	 *     <li>System function</li>
+	 *     <li>Temporary catalog function</li>
+	 *     <li>Catalog function</li>
+	 * </ul>
+	 *
+	 * @see TableEnvironment#useCatalog(String)
+	 * @see TableEnvironment#useDatabase(String)
+	 * @see TableEnvironment#createTemporaryFunction
+	 * @see TableEnvironment#createTemporarySystemFunction
+	 */
+	public static ApiExpression call(String path, Object... params) {
+		return new ApiExpression(ApiExpressionUtils.lookupCall(
+			path,
+			Arrays.stream(params).map(ApiExpressionUtils::objectToExpression).toArray(Expression[]::new)));
+	}
+
+	/**
+	 * A call to an unregistered, inline function. For functions that have been registered before and
+	 * are identified by a name, use {@link #call(String, Object...)}.
+	 *
+	 * <p><b>NOTE:</b>This call uses the new type inference stack. It means it will use the result of
```

Comment:
nit: I don't know if we should add this comment. I think it is better suited in the website docs. The JavaDocs of the scalar/table function are also updated already.